src/lib.rs

```diff
@@ -50,18 +51,43 @@ pub fn executables() -> Result<Vec<Executable>, VarError> {
     Ok(executables)
 }
 
-/// Handles precedence i.e. an entry later in PATH will override an earlier one
+#[derive(Debug, Clone, Eq)]
+struct UniqueExecutable(Executable);
+
+impl PartialEq for UniqueExecutable {
+    fn eq(&self, other: &Self) -> bool {
+        self.0.name == other.0.name
+    }
+}
+
+impl Hash for UniqueExecutable {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        self.0.name.hash(state);
+    }
+}
+
+/// Handles precedence i.e. will only return the first entry in PATH for each executable name
 ///
 /// **Note:** this will never use rayon
-pub fn unique_executables() -> Result<HashMap<String, Executable>, VarError> {
+///
+/// Only partially implemented on Windows. It will choose the correct folder but is not guaranteed
+/// to choose the correct executable based on PATHEXT order (e.g. `.exe` before `.bat` when they
+/// share a folder).
+pub fn unique_executables() -> Result<Vec<Executable>, VarError> {
```

Comment:
I return a vec, because it didn't make sense to me to make a struct public that has an unintuitive eq/hash behavior.

Reply:
Good call.